bundler/spec/bundler/source/git_spec.rb

```diff
@@ -24,5 +24,31 @@
         expect(subject.to_s).to eq "https://x-oauth-basic@github.com/foo/bar.git"
       end
     end
+
+    context "when the source has a glob specifier" do
+      let(:glob) { "bar/baz/*.gemspec" }
+      let(:options) do
+        { "uri" => uri, "glob" => glob }
+      end
+
+      it "includes it" do
+        expect(subject.to_s).to eq "https://github.com/foo/bar.git (glob: bar/baz/*.gemspec)"
+      end
+    end
+
+    context "when the source has a reference" do
```

Comment:
this was not tested, and I spent some time before realizing there was an exception being swallowed, so I added a spec.